pkg/splunk/enterprise/util.go

```diff
@@ -332,6 +332,18 @@ func ApplyAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.Me
 	appListingConfigMap.SetOwnerReferences(append(appListingConfigMap.GetOwnerReferences(), splcommon.AsOwner(cr, true)))
 
 	if len(appListingConfigMap.Data) > 0 {
+		if appsModified {
```

Comment:
I think the check @ line 334 does the same thing, and you may not need this extra flag.

Reply:
I thought so too and initially tried without the flag.  It doesn't work because if there are apps pending to be installed, the `len(Data)` will be > 0 every reconcile.  So we will reset the configMap each time causing multiple restarts.  The `appModified` flag will only be set to true when the `GetList()` of the app sources differs from what the operator has, regardless of the deployment state of the apps.  So this check is needed.

Reply:
I think, from day one, we didn't want the App framework code to be re-entrant, otherwise, we will be messing up the state/status info.  If we are entering this code across the reconciliations, wondering if we have any other bug that popping up now :-)

Reply:
I don't think so.  The `appListing` is build from the app status for any app that has yet to be fully installed.  This makes since because if the Pod gets reset for some reason prior to ansible finishing up the install, we need that list to persist so they can be still be installed.  
The `appModified` flag is a trigger to tell us to reset the configmap for the `appListing`, particularly for the case where the app set to be installed is the same as before, just the app packages have been modified.  This would only get triggered once when our S3 client code identifies a change in the app packages and updates its state.  We only need this once since if a restart/failure happens and the apps are not fully installed, the ConfigMap will still be correct.

Reply:
If we are entering multiple times, then, it may cause some issues particularly from the state machine point of view. Pod template doesn't look at `appListing`, and just uses the configMap If exists. 

As that is a separate issue, let us discuss it tomorrow, and see if we really need a fix for the same.

Reply:
CSPL-1311 created to discuss this further